tests/test_incremental.py

```diff
@@ -29,23 +30,34 @@ def test_set_params():
     assert result["scoring"] == "accuracy"
 
 
-def test_incremental_basic(scheduler):
+@pytest.mark.parametrize("dataframes", [False, True])
+def test_incremental_basic(scheduler, dataframes):
     # Create observations that we know linear models can recover
     n, d = 100, 3
     rng = da.random.RandomState(42)
     X = rng.normal(size=(n, d), chunks=30)
     coef_star = rng.uniform(size=d, chunks=d)
     y = da.sign(X.dot(coef_star))
     y = (y + 1) / 2
+    if dataframes:
+        X = dd.from_array(X)
+        y = dd.from_array(y)
 
     with scheduler() as (s, [_, _]):
         est1 = SGDClassifier(random_state=0, tol=1e-3, average=True)
         est2 = clone(est1)
 
         clf = Incremental(est1, random_state=0)
         result = clf.fit(X, y, classes=[0, 1])
-        for slice_ in da.core.slices_from_chunks(X.chunks):
-            est2.partial_fit(X[slice_], y[slice_[0]], classes=[0, 1])
+        for slice_ in da.core.slices_from_chunks(
+            X.chunks if not dataframes else X.values.chunks
+        ):
+            _X, _y = (X[slice_], y[slice_[0]]) if not dataframes else (X, y)
+            est2.partial_fit(_X, _y, classes=[0, 1])
+            if dataframes:
+                with pytest.raises(ValueError, match="cannot convert float NaN to int"):
```

Comment:
What is the cause of the value error? I don't think we want to enshrine that behavior in a test.

Reply:
> enshrine that behavior in a test.

Good point. Resolved.